statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,240 @@
+
+
+class MANOVA(Model):
+    """
+    Multivariate analysis of variance
+
+
+    Parameters
+    ----------
+    endog : array-like
+        Dependent variables (DV). A n_sample x n_y_var array where n_sample is
+        the number of observations and n_y_var is the number of DV.
+
+    exog : array-like
+        Independent variables (IV). A n_sample x n_x_var array where n is the
+        number of observations and n_x_var is the number of IV. An intercept is
+        not included by default and should be added by the user (models
+        specified using a formula include an intercept by default)
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    Attributes
+    -----------
+    df_resid : float
+        The number of observation `n` minus the number of IV `q`.
+    sscpr : array
+        Sums of squares and cross-products of residuals
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+
+    """
+    def __init__(self, endog, exog, design_info=None, **kwargs):
+        self.design_info = design_info
+        super(MANOVA, self).__init__(endog, exog)
+        self.fittedmod = manova_fit(self.exog, self.endog)
+
+    @classmethod
+    def from_formula(cls, formula, data, subset=None, drop_cols=None,
+                     *args, **kwargs):
+        mod = super(MANOVA, cls).from_formula(formula, data,
+                                              subset=subset,
+                                              drop_cols=drop_cols,
+                                              *args, **kwargs)
+        return mod
+
+    def test(self, hypothesis=None):
+        """
+        Testing the genernal hypothesis
+            L * params * M = 0
+        where `params` is the regression coefficient matrix for the
+        linear model y = x * params
+
+        Parameters
+        ----------
+        hypothesis: A list of array-like
+           Hypothesis to be tested. Each element is an array-like [name, L, M]
+           containing a string `name`, the contrast matrix L and the transform
+           matrix M for transforming dependent variables, respectively. If M is
+           `None`, it is set to an identity matrix (i.e. no dependent
+           variable transformation).
+           If `hypothesis` is None: 1) the effect of each independent variable
+           on the dependent variables will be tested. Or 2) if model is created
+           using a formula,  `hypothesis` will be created according to
+           `design_info`. 1) and 2) is equivalent if no additional variables
+           are created by the formula (e.g. dummy variables for categorical
+           variables and interaction terms)
+
+        Returns
+        -------
+        results: MANOVAResults
+
```

Comment:
@josef-pkt Do you mean create a reusable dostring for explaining the `hypotheses` argument? May I ask how to implement it?